shared/http/client.go

```diff
@@ -0,0 +1,103 @@
+package http
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+)
+
+type Client struct {
+	HTTPClient *http.Client
+	GetToken   func() (string, error) // Function to get or refresh the token
```

Comment:
```suggestion
	GetToken   func() (???, error) // Function to get or refresh the token
```
better return type possible? And then maybe nils from errors instead of empty string